File: e2e/flux/swift/Sources/Views/ProfileView.swift
// ProfileView — user profile page.

import SwiftUI

struct ProfileView: View {
    @EnvironmentObject var store: FluxStore
    let userId: String

    private var profile: ProfilePage? {
        store.get("profile/\(userId)")
    }

    var body: some View {
        Group {
            if let p = profile {
                ScrollView {
                    VStack(spacing: 16) {
                        profileHeader(p)

                        Divider()

                        if p.tweets.isEmpty {
                            Text(store.t("ui/profile/no_tweets"))
                                .foregroundColor(.secondary)
                                .padding(.top, 32)
                        } else {
                            LazyVStack(spacing: 0) {
                                ForEach(p.tweets, id: \.tweetId) { item in
                                    TweetRow(item: item)
                                        .padding(.horizontal)
                                        .padding(.vertical, 8)
                                    Divider()
                                }
                            }
                        }
                    }
                }
            } else {
                ProgressView(store.t("ui/common/loading"))
            }
        }
        .navigationTitle("@\(userId)")
        .onAppear {
            store.emit("profile/load", json: ["userId": userId])
        }
    }

    @ViewBuilder
    private func profileHeader(_ p: ProfilePage) -> some View {
        VStack(spacing: 12) {
            Circle()
                .fill(Color.blue.opacity(0.2))
                .frame(width: 72, height: 72)
                .overlay(
                    Text(String(p.user.displayName.prefix(1)))
                        .font(.title)
                        .foregroundColor(.blue)
                )

            Text(p.user.displayName)
                .font(.title2.bold())

            Text("@\(p.user.username)")
                .font(.subheadline)
                .foregroundColor(.secondary)

            if let bio = p.user.bio {
                Text(bio)
                    .font(.body)
                    .multilineTextAlignment(.center)
                    .padding(.horizontal)
            }

            HStack(spacing: 24) {
                stat(count: p.user.followerCount, label: store.t("ui/profile/followers"))
                stat(count: p.user.followingCount, label: store.t("ui/profile/following"))
                stat(count: p.user.tweetCount, label: store.t("ui/profile/tweets"))
            }

            if p.followedByMe {
                Button(action: toggleFollow) {
                    Text(store.t("ui/profile/unfollow"))
                        .frame(width: 120)
                }
                .buttonStyle(.bordered)
            } else {
                Button(action: toggleFollow) {
                    Text(store.t("ui/profile/follow"))
                        .frame(width: 120)
                }
                .buttonStyle(.borderedProminent)
            }
        }
        .padding()
    }

    private func stat(count: Int, label: String) -> some View {
        VStack {
            Text("\(count)")
                .font(.headline)
            Text(label)
                .font(.caption)
                .foregroundColor(.secondary)
        }
    }

    private func toggleFollow() {
        if profile?.followedByMe == true {
            store.emit("user/unfollow", json: ["userId": userId])
        } else {
            store.emit("user/follow", json: ["userId": userId])
        }
        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
            store.emit("profile/load", json: ["userId": userId])
        }
    }
}

File: e2e/flux/flutter/lib/views/profile_view.dart
/// ProfileView — user profile page (mirrors Swift ProfileView).
library;

import 'package:flutter/cupertino.dart';

import '../models/models.dart';
import '../store/flux_store.dart';
import 'widgets/tweet_row.dart';

class ProfileView extends StatelessWidget {
  final String userId;
  const ProfileView({super.key, required this.userId});

  @override
  Widget build(BuildContext context) {
    final store = FluxStoreScope.of(context);
    final profile = store.get<ProfilePage>('profile/$userId');

    return CupertinoPageScaffold(
      navigationBar: CupertinoNavigationBar(middle: Text('@$userId')),
      child: SafeArea(child: _buildBody(context, store, profile)),
    );
  }

  Widget _buildBody(
    BuildContext context,
    FluxStore store,
    ProfilePage? profile,
  ) {
    if (profile == null) {
      return Center(
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            const CupertinoActivityIndicator(),
            const SizedBox(height: 8),
            Text(
              store.t('ui/common/loading'),
              style: const TextStyle(color: CupertinoColors.secondaryLabel),
            ),
          ],
        ),
      );
    }

    return SingleChildScrollView(
      child: Column(
        children: [
          _profileHeader(context, store, profile),
          Container(
            height: 1,
            color: CupertinoColors.separator.resolveFrom(context),
          ),
          if (profile.tweets.isEmpty)
            Padding(
              padding: const EdgeInsets.only(top: 32),
              child: Text(
                store.t('ui/profile/no_tweets'),
                style: const TextStyle(color: CupertinoColors.secondaryLabel),
              ),
            )
          else
            ...profile.tweets.map(
              (item) => Column(
                children: [
                  Padding(
                    padding: const EdgeInsets.symmetric(
                      horizontal: 16,
                      vertical: 8,
                    ),
                    child: TweetRow(item: item),
                  ),
                  Container(
                    height: 1,
                    color: CupertinoColors.separator.resolveFrom(context),
                  ),
                ],
              ),
            ),
        ],
      ),
    );
  }

  Widget _profileHeader(
    BuildContext context,
    FluxStore store,
    ProfilePage profile,
  ) {
    final user = profile.user;
    return Padding(
      padding: const EdgeInsets.all(16),
      child: Column(
        children: [
          // Avatar
          Container(
            width: 72,
            height: 72,
            decoration: BoxDecoration(
              color: CupertinoColors.activeBlue.withAlpha(51),
              shape: BoxShape.circle,
            ),
            alignment: Alignment.center,
            child: Text(
              user.displayName.isNotEmpty ? user.displayName[0] : '?',
              style: const TextStyle(
                fontSize: 28,
                color: CupertinoColors.activeBlue,
              ),
            ),
          ),

          const SizedBox(height: 12),

          Text(
            user.displayName,
            style: const TextStyle(fontSize: 22, fontWeight: FontWeight.bold),
          ),

          const SizedBox(height: 4),

          Text(
            '@${user.username}',
            style: const TextStyle(
              fontSize: 15,
              color: CupertinoColors.secondaryLabel,
            ),
          ),

          if (user.bio != null && user.bio!.isNotEmpty) ...[
            const SizedBox(height: 8),
            Text(
              user.bio!,
              textAlign: TextAlign.center,
              style: const TextStyle(fontSize: 17),
            ),
          ],

          const SizedBox(height: 12),

          // Stats
          Row(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              _stat(user.followerCount, store.t('ui/profile/followers')),
              const SizedBox(width: 24),
              _stat(user.followingCount, store.t('ui/profile/following')),
              const SizedBox(width: 24),
              _stat(user.tweetCount, store.t('ui/profile/tweets')),
            ],
          ),

          const SizedBox(height: 12),

          // Follow / Unfollow button
          if (profile.followedByMe)
            CupertinoButton(
              onPressed: () {
                store.emit('user/unfollow', {'userId': userId});
              },
              child: SizedBox(
                width: 120,
                child: Center(child: Text(store.t('ui/profile/unfollow'))),
              ),
            )
          else
            CupertinoButton.filled(
              onPressed: () {
                store.emit('user/follow', {'userId': userId});
              },
              child: SizedBox(
                width: 120,
                child: Center(child: Text(store.t('ui/profile/follow'))),
              ),
            ),
        ],
      ),
    );
  }

  Widget _stat(int count, String label) {
    return Column(
      children: [
        Text(
          '$count',
          style: const TextStyle(fontSize: 17, fontWeight: FontWeight.w600),
        ),
        Text(
          label,
          style: const TextStyle(
            fontSize: 12,
            color: CupertinoColors.secondaryLabel,
          ),
        ),
      ],
    );
  }
}
